api/src/pcapi/core/users/external/user_automations.py

```diff
@@ -191,10 +192,31 @@ def get_users_whose_credit_expired_today() -> List[User]:
     )
 
 
+def get_ex_underage_beneficiaries_who_can_no_longer_recredit() -> List[User]:
+    # No need to be updated on the exact birthday date, but ensure that we don't miss users born one day in leap years.
+    days_19y = ceil(365.25 * 19)
+    return (
+        db.session.query(User)
+        .filter(User.has_underage_beneficiary_role.is_(True))  # type: ignore [attr-defined]
+        .filter(
+            and_(
+                User.dateOfBirth
+                > datetime.combine(date.today() - relativedelta(days=days_19y + 1), datetime.min.time()),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ça ne marche pas de passer 19 ans + 1 jour directement à `relativedelta` ?‏
```suggestion
                > datetime.combine(date.today() - relativedelta(years=19, days=1), datetime.min.time()),
```

Reply:
19 ans + 1 jour ce n'est pas continu dans le temps.

19 ans après une année bissextile, une journée ne sera pas traitée, il manque dans les résultats la naissance le 28/02/2004 sur cet exemple en 2023 (ça va venir vite !) : 
```
>>> datetime(2023, 2, 26) - relativedelta(years=19, days=1)
datetime.datetime(2004, 2, 25, 0, 0)
>>> datetime(2023, 2, 27) - relativedelta(years=19, days=1)
datetime.datetime(2004, 2, 26, 0, 0)
>>> datetime(2023, 2, 28) - relativedelta(years=19, days=1)
datetime.datetime(2004, 2, 27, 0, 0)
>>> datetime(2023, 3, 1) - relativedelta(years=19, days=1)
datetime.datetime(2004, 2, 29, 0, 0)
>>> datetime(2023, 3, 2) - relativedelta(years=19, days=1)
datetime.datetime(2004, 3, 1, 0, 0)
```

À l'inverse, les années bissextiles, on traitera deux fois la même journée de naissance, ce second cas n'est pas vraiment gênant puisqu'on mettrait juste à jour deux fois : 
```
>>> datetime(2024, 2, 27) - relativedelta(years=19, days=1)
datetime.datetime(2005, 2, 26, 0, 0)
>>> datetime(2024, 2, 28) - relativedelta(years=19, days=1)
datetime.datetime(2005, 2, 27, 0, 0)
>>> datetime(2024, 2, 29) - relativedelta(years=19, days=1)
datetime.datetime(2005, 2, 27, 0, 0)
>>> datetime(2024, 3, 1) - relativedelta(years=19, days=1)
datetime.datetime(2005, 2, 28, 0, 0)
>>> datetime(2024, 3, 2) - relativedelta(years=19, days=1)
datetime.datetime(2005, 3, 1, 0, 0)
```